### ISSUE TITLE
Error in next_message_reader

### ISSUE THREAD

**Issue by Roman-Koshelev** [jamesdbrock/hffix#36]
The line "8=FIX" will not necessarily be the beginning of a new message. For example, you will get "58=FIX arbitrary string". Search correctly for "\18=FIX"

**Comment by jamesdbrock** [comment]
Yeah, good point. That's a rare case but it could occur. That should be fixed. Do you want to make a PR?

**Comment by Roman-Koshelev** [comment]
Yes, I will be happy to do PR. (a couple of hours and it's all in the bag)
